### PR TITLE
[GAM] Converted GAM to Python module.

### PR DESCRIPTION
**Pull request by otiliastr**
Converted GAM to a Python module and updated readme with running instructions.

@kvis-google @dattias

### CODE REVIEW

File: research/gam/README.md [hunk 1]
```diff
@@ -17,26 +17,46 @@ likely to have the same label, thus guiding its parameters to a better local
 optima. The classification and agreement models are trained jointly in a
 co-training fashion.
 
-This code repository contains an implementation of Graph Agreement Models. The
-code is organized into the following folders:
+This code repository contains an implementation of Graph Agreement Models [3].
```
Comment (kvis-google) [inline research/gam/README.md]: Can we move this to the beginning of the write-up. The reference number would also have to change to [1].

File: research/gam/README.md [hunk 2]
```diff
@@ -17,26 +17,46 @@ likely to have the same label, thus guiding its parameters to a better local
...
+
 ## References
 
 [[1] T. Bui, S. Ravi and V. Ramavajjala. "Neural Graph Learning: Training Neural
 Networks Using Graphs." WSDM 2018](https://ai.google/research/pubs/pub46568.pdf)
 
 [[2] T. Kipf and M. Welling. "Semi-supervised classification with graph
 convolutional networks." ICLR 2017](https://arxiv.org/pdf/1609.02907.pdf)
+
+[[3] O. Stretcu, K. Viswanathan, D. Movshovitz-Attias, E.A. Platanios,
```
Comment (kvis-google) [inline research/gam/README.md]: Please move this to be the first reference if we address the comment further up.

File: research/gam/README.md [hunk 3]
```diff
@@ -17,26 +17,46 @@ likely to have the same label, thus guiding its parameters to a better local
...
+*   experiments: Python run script for training Graph Agreement Models on
+    CIFAR10 and other datasets.
 
 The implementations of Graph Agreement Models (GAMs) are provided in the `gam`
 folder on a strict "as is" basis, without warranties or conditions of any kind.
 Also, these implementations may not be compatible with certain TensorFlow
 versions (such as 2.0 or above) or Python versions.
 
+## How to run
+To run GAM on a graph-based dataset (e.g., Cora, Citeseer), from this folder
```
Comment (kvis-google) [inline research/gam/README.md]: Please add Pubmed to list.

File: research/gam/README.md (hunk 1 above)
Comment (otiliastr) [reply research/gam/README.md]: Done.

File: research/gam/README.md (hunk 2 above)
Comment (otiliastr) [reply research/gam/README.md]: Done.

**Pull request merged**